packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -731,6 +766,32 @@ class Actions extends StatefulWidget {
   /// {@endtemplate}
   final Map<Type, Action<Intent>> actions;
 
+  /// {@template flutter.widgets.actions.registry}
+  /// A registry of actions to be added to the list of actions to be searched.
+  ///
+  /// The action mappings registered with the registry will be searched in
+  /// addition to the [actions] map, and can be added to by creating a
+  /// [RegisteredActions] widget as a descendant of this [Actions] widget.
+  /// {@endtemplate}
+  final ActionRegistry? registry;
+
+  /// Register the actions here with the nearest ancestor with a [registry] set.
+  ///
+  /// Actions defined in this widget will not be registered with this widget's
+  /// [registry], only with the nearest ancestor [registry].
+  ///
+  /// These registered actions will be searched when invoking an intent in any
+  /// context below the [Actions] widget that defines the registry. So, if the
+  /// registry is defined at the root of the widget hierarchy, then these actions
+  /// are defined globally, and invoking them with their registered intents will
+  /// invoke them. The [WidgetsApp] (from which [MaterialApp] and [CupertinoApp]
+  /// are derived) defines a registry, so if you use one of these, a registry will
+  /// already be defined at the top level.
+  ///
+  /// If more than one action is registered as responding to the same [Intent]
+  /// type, all of the actions will be invoked.
```

Comment:
Is this still true even if I have a local Actions widget further down the tree that binds the same intent? Will the registered actions always fire? Is there a way to disable them for subtrees further down?

Reply:
Yeah, at the moment, it will. The way I was thinking you could control this was through either the `isEnabled` for the action, through placing a `ActionsRegistry` widget to collect registrations from subtrees you wanted to restrict, and through just not adding the action to the action map if a condition was true.  That might not be enough, though, since you might not control any of those things if you're using someone else's widget.